.editorconfig

```diff
@@ -0,0 +1,188 @@
+# To learn more about .editorconfig see https://aka.ms/editorconfigdocs
+# editorconfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Default settings:
+# A newline ending every file
+# Use 4 spaces as indentation
+[*]
+insert_final_newline = true
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+
+[project.json]
+indent_size = 2
+
+# C# files
+[*.cs]
+# New line preferences
+csharp_new_line_before_open_brace = all
+csharp_new_line_before_else = true
```

Comment:
I don't see this rule enforced, e.g. the file from this branch wasn't formatted for some reason:
[Decimal128.GetDecimal128Type](https://github.com/rstam/mongo-csharp-driver/blob/fbb4f537c491c85a97d19e1ede27bbd07fa6d22e/src/MongoDB.Bson/ObjectModel/Decimal128.cs#L1845)
This is strange, since `dotnet-format` claims to support this option:
[Supported C# formatting settings](https://github.com/dotnet/format/blob/master/docs/Supported-.editorconfig-options.md#c-formatting-settings)

Reply:
I think GetDecimal128Type DOES adhere to `csharp_new_line_before_else`.

Look at the C# code... each `else` is preceded by a new line.

Reply:
Yes, indeed. For some reason I expected it to add a new line after as well.